syft/file/locations.go

```diff
@@ -33,3 +37,52 @@ func (l Locations) Less(i, j int) bool {
 func (l Locations) Swap(i, j int) {
 	l[i], l[j] = l[j], l[i]
 }
+
+type locationsByContainerOrder struct {
+	locations          []Location
+	layerOrderByDigest map[string]int
+}
+
+func LocationsByContainerOrder(locations []Location, layerOrderByDigest map[string]int) sort.Interface {
```

Comment:
Could this be internal? The `map[string]int` seems a bit brittle for a public API here.